alvr/server_openvr/cpp/platform/win32/FrameRender.cpp

```diff
@@ -585,39 +691,123 @@ bool FrameRender::RenderFrame(
         // Update uv-coordinates in vertex buffer according to bounds.
         //
 
+        DirectX::XMMATRIX framePose
+            = (i == recenterLayer) ? identityMat : HmdMatrix_AsDxMatOrientOnly(poses[i]);
+        DirectX::XMMATRIX framePoseInv = DirectX::XMMatrixInverse(nullptr, framePose);
+
+        // framePose is the position of the layer in space, ie an identity matrix
+        // would place the quad perpendicular in the floor at 0,0,0
+        DirectX::XMMATRIX viewMatDiff
+            = DirectX::XMMatrixInverse(nullptr, hmdPoseForTargetTs * framePoseInv);
+
+        DirectX::XMMATRIX transformMatL = viewMatDiff * hmdToEyeMatL * projectionMatL;
+        DirectX::XMMATRIX transformMatR = viewMatDiff * hmdToEyeMatR * projectionMatR;
+
+        if (i == recenterLayer) {
+            transformMatL = identityMat;
+            transformMatR = identityMat;
+        }
+
+        const auto depth = 700.0f;
+        const auto m = 1.0f;
+        DirectX::XMFLOAT4 vertsL[4];
+        DirectX::XMFLOAT4 vertsR[4];
+
+        // Debugging quad if 0, useful for sorting out issues
+#if 1
+        DirectX::XMVECTORF32 vertsL_VF32[4]
+            = { { { { -1.0f * -viewProj[0].vTopLeft.v[0] * depth * m,
+                      1.0f * -viewProj[0].vTopLeft.v[1] * depth * m,
+                      -depth,
+                      1.0f } } },
+                { { { 1.0f * viewProj[0].vBottomRight.v[0] * depth * m,
+                      -1.0f * viewProj[0].vBottomRight.v[1] * depth * m,
+                      -depth,
+                      1.0f } } },
+                { { { 1.0f * viewProj[0].vBottomRight.v[0] * depth * m,
+                      1.0f * -viewProj[0].vTopLeft.v[1] * depth * m,
+                      -depth,
+                      1.0f } } },
+                { { { -1.0f * -viewProj[0].vTopLeft.v[0] * depth * m,
+                      -1.0f * viewProj[0].vBottomRight.v[1] * depth * m,
+                      -depth,
+                      1.0f } } } };
+#else
+        DirectX::XMVECTORF32 vertsL_VF32[4]
+            = { { { { -1.0f * depth * m, 1.0f * depth * m, -depth, 1.0f } } },
+                { { { 1.0f * depth * m, -1.0f * depth * m, -depth, 1.0f } } },
+                { { { 1.0f * depth * m, 1.0f * depth * m, -depth, 1.0f } } },
+                { { { -1.0f * depth * m, -1.0f * depth * m, -depth, 1.0f } } } };
+#endif
```

Comment:
Why did you hardcode disable this code with directives? Should it be removed?

Reply:
could remove it yeah, wasn't sure if I'd have to come back to this code or not